opbeat/contrib/django/client.py

```diff
@@ -65,46 +66,30 @@ def __init__(self, **kwargs):
         super(DjangoClient, self).__init__(**kwargs)
 
     def get_user_info(self, request):
+        user_info = {}
+
+        if not hasattr(request, 'user'):
+            return user_info
         try:
-            if request.user.is_authenticated():
-                user_info = {
-                    'is_authenticated': True,
-                    'id': request.user.pk,
-                }
-                if hasattr(request.user, 'get_username'):
-                    user_info['username'] = request.user.get_username()
-                elif hasattr(request.user, 'username'):
-                    user_info['username'] = request.user.username
-                else:
-                    # this only happens if the project uses custom user models, but
-                    # doesn't correctly inherit from AbstractBaseUser
-                    user_info['username'] = ''
-
-                if hasattr(request.user, 'email'):
-                    user_info['email'] = request.user.email
-            else:
-                user_info = {
-                    'is_authenticated': False,
-                }
+            user = request.user
+            if hasattr(user, 'is_authenticated'):
+                user_info['is_authenticated'] = user.is_authenticated()
+            if hasattr(user, 'id'):
+                user_info['id'] = user.id
+            if hasattr(user, 'get_username'):
+                user_info['username'] = user.get_username()
```

Comment:
theoretically yes, but practically the user has been fetched by other code at this point.

It might be a good idea to check if the `user` object is still a `LazyObject` at this point, which would indicate that the user data has not been fetched from the DB yet. I'll make a note to look into it